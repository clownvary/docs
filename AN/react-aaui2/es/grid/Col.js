import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import React from 'react';
import classnames from 'classnames';
import { number, string, node } from 'prop-types';

var propTypes = {
  className: string,
  children: node,
  span: number,
  sm: number,
  md: number,
  lg: number
};
var defaultProps = {
  span: 12
};

export default function Col(_ref) {
  var _classnames;

  var children = _ref.children,
      span = _ref.span,
      sm = _ref.sm,
      md = _ref.md,
      lg = _ref.lg,
      className = _ref.className,
      rest = _objectWithoutProperties(_ref, ['children', 'span', 'sm', 'md', 'lg', 'className']);

  return React.createElement(
    'div',
    _extends({
      className: classnames('col', (_classnames = {}, _classnames['col-' + span] = span, _classnames['col-sm-' + sm] = sm, _classnames['col-md-' + md] = md, _classnames['col-lg-' + lg] = lg, _classnames), className)
    }, rest),
    children
  );
}

Col.propTypes = propTypes;
Col.defaultProps = defaultProps;