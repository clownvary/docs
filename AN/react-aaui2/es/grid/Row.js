import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import React from 'react';
import classNames from 'classnames';
import { oneOf, bool, number, string, object, node, oneOfType } from 'prop-types';

var propsTypes = {
  className: string,
  fluid: bool,
  gutter: oneOfType([string, number, bool]),
  align: oneOf(['start', 'center', 'end', 'stretch']),
  justify: oneOf(['start', 'center', 'end', 'between', 'around']),
  style: object,
  children: node
};

var defaultProps = {
  fluid: false,
  gutter: true,
  align: 'start',
  justify: 'start',
  style: {}
};

export default function Row(props) {
  var _classNames;

  var fluid = props.fluid,
      justify = props.justify,
      align = props.align,
      className = props.className,
      gutter = props.gutter,
      style = props.style,
      children = props.children,
      rest = _objectWithoutProperties(props, ['fluid', 'justify', 'align', 'className', 'gutter', 'style', 'children']);

  var justifyPrefix = justify === 'between' || justify === 'around' ? 'space-' + justify : justify;

  var boolGutter = typeof gutter === 'boolean';
  var classes = classNames((_classNames = {
    row: !fluid,
    'row-fluid': fluid,
    'row-gutter': boolGutter && !!gutter
  }, _classNames['row-align-' + align] = align, _classNames['row-justify-' + justifyPrefix] = justify, _classNames), className);

  // prettier-ignore
  var rowStyle = !boolGutter && gutter > 0 ? _extends({
    marginLeft: gutter / -2,
    marginRight: gutter / -2
  }, style) : style;
  var cols = React.Children.map(children, function (col) {
    if (!boolGutter && col && col.props && gutter > 0) {
      return React.cloneElement(col, {
        style: _extends({
          paddingLeft: gutter / 2,
          paddingRight: gutter / 2
        }, col.props.style)
      });
    }
    return col;
  });

  return React.createElement(
    'div',
    _extends({ className: classes, style: rowStyle }, rest),
    cols
  );
}

Row.propTypes = propsTypes;
Row.defaultProps = defaultProps;