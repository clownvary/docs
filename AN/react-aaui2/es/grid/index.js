import _Row from './Row';
export { _Row as Row };
import _Col from './Col';
export { _Col as Col };