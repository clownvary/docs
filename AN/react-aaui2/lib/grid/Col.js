'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

exports.default = Col;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _propTypes = require('prop-types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  className: _propTypes.string,
  children: _propTypes.node,
  span: _propTypes.number,
  sm: _propTypes.number,
  md: _propTypes.number,
  lg: _propTypes.number
};
var defaultProps = {
  span: 12
};

function Col(_ref) {
  var _classnames;

  var children = _ref.children,
      span = _ref.span,
      sm = _ref.sm,
      md = _ref.md,
      lg = _ref.lg,
      className = _ref.className,
      rest = (0, _objectWithoutProperties3.default)(_ref, ['children', 'span', 'sm', 'md', 'lg', 'className']);

  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({
      className: (0, _classnames3.default)('col', (_classnames = {}, _classnames['col-' + span] = span, _classnames['col-sm-' + sm] = sm, _classnames['col-md-' + md] = md, _classnames['col-lg-' + lg] = lg, _classnames), className)
    }, rest),
    children
  );
}

Col.propTypes = propTypes;
Col.defaultProps = defaultProps;
module.exports = exports['default'];