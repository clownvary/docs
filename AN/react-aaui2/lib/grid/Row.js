'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

exports.default = Row;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _propTypes = require('prop-types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propsTypes = {
  className: _propTypes.string,
  fluid: _propTypes.bool,
  gutter: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.number, _propTypes.bool]),
  align: (0, _propTypes.oneOf)(['start', 'center', 'end', 'stretch']),
  justify: (0, _propTypes.oneOf)(['start', 'center', 'end', 'between', 'around']),
  style: _propTypes.object,
  children: _propTypes.node
};

var defaultProps = {
  fluid: false,
  gutter: true,
  align: 'start',
  justify: 'start',
  style: {}
};

function Row(props) {
  var _classNames;

  var fluid = props.fluid,
      justify = props.justify,
      align = props.align,
      className = props.className,
      gutter = props.gutter,
      style = props.style,
      children = props.children,
      rest = (0, _objectWithoutProperties3.default)(props, ['fluid', 'justify', 'align', 'className', 'gutter', 'style', 'children']);

  var justifyPrefix = justify === 'between' || justify === 'around' ? 'space-' + justify : justify;

  var boolGutter = typeof gutter === 'boolean';
  var classes = (0, _classnames2.default)((_classNames = {
    row: !fluid,
    'row-fluid': fluid,
    'row-gutter': boolGutter && !!gutter
  }, _classNames['row-align-' + align] = align, _classNames['row-justify-' + justifyPrefix] = justify, _classNames), className);

  // prettier-ignore
  var rowStyle = !boolGutter && gutter > 0 ? (0, _extends3.default)({
    marginLeft: gutter / -2,
    marginRight: gutter / -2
  }, style) : style;
  var cols = _react2.default.Children.map(children, function (col) {
    if (!boolGutter && col && col.props && gutter > 0) {
      return _react2.default.cloneElement(col, {
        style: (0, _extends3.default)({
          paddingLeft: gutter / 2,
          paddingRight: gutter / 2
        }, col.props.style)
      });
    }
    return col;
  });

  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({ className: classes, style: rowStyle }, rest),
    cols
  );
}

Row.propTypes = propsTypes;
Row.defaultProps = defaultProps;
module.exports = exports['default'];