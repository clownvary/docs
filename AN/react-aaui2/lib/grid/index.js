'use strict';

exports.__esModule = true;
exports.Col = exports.Row = undefined;

var _Row2 = require('./Row');

var _Row3 = _interopRequireDefault(_Row2);

var _Col2 = require('./Col');

var _Col3 = _interopRequireDefault(_Col2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.Row = _Row3.default;
exports.Col = _Col3.default;