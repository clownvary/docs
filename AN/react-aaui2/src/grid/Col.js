import React from 'react'
import classnames from 'classnames'
import { number, string, node } from 'prop-types'

const propTypes = {
  className: string,
  children: node,
  span: number,
  sm: number,
  md: number,
  lg: number,
}
const defaultProps = {
  span: 12,
}

export default function Col({
  children,
  span,
  sm,
  md,
  lg,
  className,
  ...rest
}) {
  return (
    <div
      className={classnames(
        'col',
        {
          [`col-${span}`]: span,
          [`col-sm-${sm}`]: sm,
          [`col-md-${md}`]: md,
          [`col-lg-${lg}`]: lg,
        },
        className,
      )}
      {...rest}
    >
      {children}
    </div>
  )
}

Col.propTypes = propTypes
Col.defaultProps = defaultProps
