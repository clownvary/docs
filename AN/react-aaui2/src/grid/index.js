export Row from './Row'
export Col from './Col'
