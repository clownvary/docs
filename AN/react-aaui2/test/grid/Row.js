import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Row, Col } from 'grid'

const baseProps = {
  fluid: false,
  gutter: 0,
  align: 'start',
  justify: 'start',
  style: {},
}

it('should render without errors', () => {
  const wrapper = shallow(<Row {...baseProps}>Test</Row>)

  expect(wrapper).toBeTruthy()
})

it("should render right thing and don't change unexpected", () => {
  const tree = renderer
    .create(
      <Row {...baseProps}>
        <div>Col</div>
      </Row>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('should render without errors when gutter greater than 0', () => {
  const wrapper = shallow(
    <Row {...baseProps} gutter={2}>
      <div>Col</div>
    </Row>,
  )

  expect(wrapper).toBeTruthy()
})

it('should render fluid rightly', () => {
  const wrapper = shallow(<Row {...baseProps} fluid />)

  const justifier = wrapper.find('.row-fluid')
  expect(justifier.length).toBe(1)

  expect(wrapper).toBeTruthy()
})

it('should render justify rightly', () => {
  const wrapper = shallow(<Row {...baseProps} justify={'around'} />)

  const justifier = wrapper.find('.row-justify-space-around')
  expect(justifier.length).toBe(1)

  expect(wrapper).toBeTruthy()
})

it('should support gutter with bool type', () => {
  const wrapper = shallow(<Row gutter />)

  expect(wrapper.hasClass('row-gutter')).toBe(true)
})

it('should support gutter with number type', () => {
  const tree = renderer
    .create(
      <Row gutter={30}>
        <Col>Col1</Col>
      </Row>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('should support gutter with string type', () => {
  const tree = renderer
    .create(
      <Row gutter={30}>
        <Col>Col1</Col>
      </Row>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
