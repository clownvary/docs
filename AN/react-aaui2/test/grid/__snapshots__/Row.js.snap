// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<div
  className="row row-align-start row-justify-start"
  style={Object {}}
>
  <div>
    Col
  </div>
</div>
`;

exports[`should support gutter with number type 1`] = `
<div
  className="row row-align-start row-justify-start"
  style={
    Object {
      "marginLeft": -15,
      "marginRight": -15,
    }
  }
>
  <div
    className="col col-12"
    style={
      Object {
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    Col1
  </div>
</div>
`;

exports[`should support gutter with string type 1`] = `
<div
  className="row row-align-start row-justify-start"
  style={
    Object {
      "marginLeft": -15,
      "marginRight": -15,
    }
  }
>
  <div
    className="col col-12"
    style={
      Object {
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    Col1
  </div>
</div>
`;
